server/chat-plugins/helptickets.ts

```diff
@@ -352,6 +352,17 @@ export class HelpTicket extends Rooms.RoomGame {
 		// @ts-ignore
 		this.playerTable = null;
 	}
+	onChatMessage(message: string, user: User) {
+		// in case they namechange
+		if (this.room.auth.has(user.id) && user.connections.length) {
+			const roomids = message.match(/battle-(?:[a-zA-Z0-9]+)-(?:[0-9]+)(?:-(?:[a-zA-Z0-9]+)pw)?/ig);
```

Comment:
```suggestion
			const roomids = message.match(/battle-(?:[a-z0-9]+)-(?:[0-9]+)(?:-[a-z0-9]+pw)?/ig);
```

1. roomids must be all-lowercase
2. you have an extraneous set of parens